extensions/positron-catalog-explorer/src/catalogs/snowflake.ts

```diff
@@ -513,76 +565,47 @@ async function generateCode(
+ * @param warehouse The warehouse name
  * @param database The database name
  * @param schema The schema name
  * @param table The table name
+ * @param connOptions Optional connections object
  * @returns Generated code and required dependencies
  */
-function getPythonCodeForSnowflakeTable(
-	accountName: string,
-	username: string,
+async function getPythonCodeForSnowflakeTable(
+	connName: string,
 	warehouse: string,
 	database?: string,
 	schema?: string,
-	table?: string
-): { code: string; dependencies: string[] } {
-	const dependencies = ['snowflake-connector-python[secure-local-storage]', 'pandas'];
-
-	let code = `# pip install ${dependencies.join(' ')}\n`;
-
-	if (database && schema && table) {
-		code += `# For ${accountName}.${database}.${schema}.${table}\n`;
-	} else {
-		code += `# For ${accountName}\n`;
-	}
-
-	code += `
-import snowflake.connector as sc
-
-conn_params = {
-	'account': '${accountName}',
-	'user': '${username}',
-	'authenticator': 'externalbrowser',
-`;
-
-	// add database if provided
-	if (database) {
-		code += `	'database': '${database}',`;
-	}
-
-	// add schema if provided
-	if (schema) {
-		code += `\n	'schema': '${schema}'`;
-	}
-
-	// complete the connection parameters dict and establish connection
-	code += `
-}
-
-# Establish the connection
-conn = sc.connect(**conn_params)
-cursor = conn.cursor()
-cursor.execute("USE WAREHOUSE ${warehouse}")
-`;
-	// add query to fetch data from the specified table or default info
-	if (table) {
-		code += `query = "SELECT * FROM ${table} LIMIT 10"`;
-	} else {
-		code += `query = "SELECT CURRENT_VERSION(), CURRENT_USER(), CURRENT_ROLE()"`;
-	}
-
-	// execute the query
-	code += `
-cursor.execute(query)
-
-# Fetch and display results
-results = cursor.fetchall()
-for row in results:
-	print(row)
-
-cursor.close()
+	table?: string,
+	connOptions?: any
+): Promise<{ code: string; dependencies: string[] }> {
+	const dependencies = ['"snowflake-connector-python[secure-local-storage, pandas]"', 'pandas'];
+	const absoluteTablePath = [database, schema, table].filter(part => part).join('.');
+	const label = absoluteTablePath ? `For ${absoluteTablePath}` : `For ${connName}`;
+
+	// Prepare SQL query based on whether a table was provided
+	const query = absoluteTablePath
+		? `SELECT * FROM ${absoluteTablePath} LIMIT 10`
+		: `SELECT CURRENT_VERSION(), CURRENT_USER(), CURRENT_ROLE()`;
+
+	// Use a single template literal for the entire Python code
+	const code = `# pip install ${dependencies.join(' ')}
+# ${label}
+
+import snowflake.connector
+
+with snowflake.connector.connect(connection_name="${connName}") as conn:
+\twith conn.cursor() as cursor:
+${connOptions?.warehouse ? '' : `\t\tcursor.execute("USE WAREHOUSE ${warehouse}")\n`}
+\t\tquery = "${query}"
+\t\t# Execute the query
+\t\tcursor.execute(query)
+\t\t# Fetch and display results
+\t\tresults = cursor.fetch_pandas_all()
+\t\tfor row in results:
+\t\t\tprint(row)
```

Comment:
When trying this code I got `No active warehouse selected in the current session.  Select an active warehouse with the 'use warehouse' command.`. Is that on me for not having it in my toml info? Or should we be able to figure that out from the tree somehow?

Reply:
Nope, it didn't show. But that could be my fault. In Snowflake I did "Connect a tool to Snowflake" > "Config File" to get a sample TOML entry, and that includes
```toml
warehouse = "<none selected>"
database = "<none selected>"
schema = "<none selected>"
```
I deleted that section and the quickpick shows up! (Not sure if there's special handling for those strings in the python library? but it's probably fine to chalk that up to user error haha)